Source/utils/file_util.cpp

```diff
@@ -472,4 +475,64 @@ FILE *OpenFile(const char *path, const char *mode)
 #endif
 }
 
+std::vector<std::string> ListDirectories(const char *path)
+{
+	std::vector<std::string> dirs;
+#ifdef NXDK
+	// Use the NXDK Windows API.
+	WIN32_FIND_DATAA findData;
+	// Construct the search path by appending the directory separator and wildcard.
+	std::string searchPath = std::string(path) + DIRECTORY_SEPARATOR_STR + "*";
+	HANDLE hFind = FindFirstFileA(searchPath.c_str(), &findData);
+	if (hFind == INVALID_HANDLE_VALUE)
+		return dirs;
+	do {
+		std::string folder = findData.cFileName;
+		// Skip the special entries "." and ".."
+		if (folder == "." || folder == "..")
+			continue;
+		if (findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY)
+			dirs.push_back(folder);
+	} while (FindNextFileA(hFind, &findData));
+	FindClose(hFind);
+#else
+	// For platforms that support std::filesystem.
+	for (const auto &entry : std::filesystem::directory_iterator(path)) {

```

Comment:
Oh, I guess the same thing about Unicode characters applies to the path that gets passed in. We have lots of places that show how you should reinterpret_cast `char` to `char8_t` when using `std::filesystem`. For example:

https://github.com/diasurgical/DevilutionX/blob/0f35bd8ce220f0c019b9a19459af90c1444d262a/Source/utils/file_util.cpp#L109

Reply:
One more thing I noticed in this example is that this uses the overload of `std::filesystem::exists()` that takes `std::error_code` to avoid throwing exceptions. We should be using similar overloads here as well.

Reply:
It doesn't look like the iterator will throw an exception: https://en.cppreference.com/w/cpp/filesystem/directory_iterator

Reply:
You have to look at the documentation for the constructor.
https://en.cppreference.com/w/cpp/filesystem/directory_iterator/directory_iterator

![image](https://github.com/user-attachments/assets/801e1640-083f-443b-87db-0189dc2d208b)